static/app/views/insights/mobile/screens/components/screensOverview.tsx

```diff
@@ -0,0 +1,235 @@
+      enabled: enabled,
+      referrer: Referrer.SCREENS_SCREEN_TABLE,
+    }),
+  };
+}
+
+export function ScreensOverview() {
+  const router = useRouter();
+  const {selection} = usePageFilters();
+  const location = useLocation();
+  const organization = useOrganization();
+  const {isProjectCrossPlatform, selectedPlatform} = useCrossPlatformProject();
+  const [visibleScreens, setVisibleScreens] = useState<string[]>([]);
+  const sortedBy = decodeScalar(location.query.sort, '-count');
+  const sortField = decodeSorts([sortedBy])[0].field;
+
+  const transactionMetricsDataset = DiscoverDatasets.METRICS;
+  const transactionMetricsFields = [
+    SpanMetricsField.PROJECT_ID,
+    SpanMetricsField.TRANSACTION,
+    `count()`,
+    'avg(measurements.app_start_cold)',
+    'avg(measurements.app_start_warm)',
+    `avg(measurements.time_to_initial_display)`,
+    `avg(measurements.time_to_full_display)`,
+  ];
+
+  const spanMetricsDataset = DiscoverDatasets.SPANS_METRICS;
+  const spanMetricsFields = [
+    SpanMetricsField.PROJECT_ID,
+    SpanMetricsField.TRANSACTION,
+    `avg(mobile.slow_frames)`,
+    `avg(mobile.frozen_frames)`,
+    `avg(mobile.frames_delay)`,
+  ];
+
+  const isSpanPrimary = spanMetricsFields.some(
+    field => getAggregateAlias(field) === sortField
+  );
+  const primaryDataset = isSpanPrimary ? spanMetricsDataset : transactionMetricsDataset;
+  const primaryFields = isSpanPrimary ? spanMetricsFields : transactionMetricsFields;
+
+  const secondaryDataset = isSpanPrimary ? transactionMetricsDataset : spanMetricsDataset;
+  const secondaryFields = isSpanPrimary ? transactionMetricsFields : spanMetricsFields;
+
+  const {
+    data: primaryData,
+    isLoading: primaryLoading,
+    pageLinks: primaryLinks,
+    eventView: primaryEventView,
+  } = useMetrics(
+    primaryDataset,
+    primaryFields,
+    isProjectCrossPlatform ? selectedPlatform : undefined,
+    selection,
+    location,
+    sortedBy,
+    true,
+    []
+  );
+
+  const {
+    data: secondaryData,
+    // isLoading: secondaryLoading,
+    // pageLinks: secondaryLinks,
+    // eventView: secondaryEventView,
+  } = useMetrics(
+    secondaryDataset,
+    secondaryFields,
+    isProjectCrossPlatform ? selectedPlatform : undefined,
+    selection,
+    location,
+    undefined,
+    visibleScreens.length > 0,
+    visibleScreens
+  );
+
+  useEffect(() => {
+    if (primaryData) {
+      const screens: string[] = new Array();
+      primaryData?.data.forEach(row => {
+        if (row.transaction) {
+          screens.push(String(row.transaction));
+        }
+      });
+      setVisibleScreens(screens);
+    }
+  }, [primaryData]);
+
+  const tableSearchFilters = new MutableSearch(['transaction.op:ui.load']);
+  const derivedQuery = getTransactionSearchQuery(location, primaryEventView.query);
+
+  const combinedData = useMemo((): TableData | undefined => {
+    if (primaryData !== undefined && secondaryData !== undefined) {
```

Comment:
nit: We have a `defined` helper that makes this kind of check look a little neater